__init__.py

```diff
@@ -140,6 +115,15 @@ def _find_entity(self, entity, domains):
                               "dev_name": entity})
         return ha_entity
 
+    # Routine for entiti availibility check
+    def _check_availability(self, ha_entity):
+        self.log.debug(ha_entity['state'])
+        if ha_entity['state'] == 'unavailable':
+            self.speak_dialog('homeassistant.error.unavailable', data={
+                            "dev_name": ha_entity['dev_name']})
```

Comment:
This dialog has no dependency on the function it exists in, meaning that it can be moved out of the function to clean it up.

The function name indicates that we will get a return value, but not that any dialog or other actions will happen. I suggest removing this dialog outside of the function and using it at the location where the value of the return is known